src/main/java/ru/hogwarts/school/service/StudentServiceImpl.java

```diff
@@ -139,4 +140,27 @@ public List<Student> findLastFiveStudent() {
         logger.info("Вызов метода для поиска последних пяти студентов");
         return studentRepository.getLastFiveStudent();
     }
+
+    @Override
+    public Collection<Student> findStudentByStartsWithA() {
+        logger.info("Вызов метода поиска студентов по имени, начинающегося на А");
+        return studentRepository.findAll()
+                .stream()
+                .filter(a -> a.getName()
+                        .toUpperCase()
+                        .startsWith("А"))
+                .sorted(Comparator.comparing(Student::getName))
+                .toList();
+    }
+
+    @Override
+    public Double findAverageStudentByAge() {
+        logger.info("Вызов метода подсчёта среднего значения возраста студентов");
+        return studentRepository.findAll()
+                .stream()
+                .map(Student::getAge)
+                .mapToInt(s -> s)
```

Comment:
Можно было одним мапом обойтись
```suggestion
                .mapToInt(s -> s.getAge())
```